src/test/java/com/ccs/inventorymanagement/ClothingServiceTest.java

```diff
@@ -0,0 +1,253 @@
+                .thenReturn(id);
+
+
+        // Then
+        clothingService.findAll()
+                .as(StepVerifier::create)
+                .expectNextCount(1)
+                .verifyComplete();
+        verify(clothingRepository, times(1)).findAll();
+    }
+
+    @Test
+    @DisplayName("GET Find By ID: Should return clothing by ID")
+    public void shouldReturnClothingById() {
+
+        // Given
+        final UUID id = UUID.randomUUID();
+        ClothingEntity entity = mock(ClothingEntity.class);
+
+        // When
+        when(clothingRepository.findById(id))
+                .thenReturn(Mono.just(entity));
+        when(entity.getId())
+                .thenReturn(id);
+
+        // Then
+        clothingService.findById(id)
+                .as(StepVerifier::create)
+                .assertNext(clothing -> assertEquals(id, clothing.getId()))
+                .verifyComplete();
+        verify(clothingRepository, times(1)).findById(id);
+    }
+
+    @Test
+    @DisplayName("GET Find By ID: Should not return clothing by ID if does not exist")
+    public void shouldNotReturnClothingIfNotFoundById() {
+
+        // Given
+        final UUID id = UUID.randomUUID();
+
+        // When
+        when(clothingRepository.findById(id))
+                .thenReturn(Mono.empty());
+
+        // Then
+        clothingService.findById(id)
+                .as(StepVerifier::create)
+                .verifyComplete(); // makes sure reactive chain completes
+        verify(clothingRepository, times(1)).findById(id);
+    }
+
+    @Test
+    @DisplayName("GET Find By ID: Should return errors")
+    public void shouldReturnErrors() {
+
+        // Given
+        final UUID id = UUID.randomUUID();
+
+        // When
+        when(clothingRepository.findById(id))
+                .thenReturn(Mono.error(new R2dbcBadGrammarException()));
+
+        // Then
+        clothingService.findById(id)
+                .as(StepVerifier::create)
+                .verifyError(R2dbcBadGrammarException.class);
+        verify(clothingRepository, times(1)).findById(id);
+    }
+
+    @Test
+    @DisplayName("PUT Update Clothing: Should update clothing by ID")
+    public void shouldUpdateClothing() {
+        //Given
+        final UUID id = UUID.randomUUID();
+        Clothing clothing = mock(Clothing.class);
+        ClothingEntity entity = ClothingEntity.builder().build();
+        Instant initialTime = Instant.now();
+        ClothingEntity saved = ClothingEntity.builder().updated(initialTime).build();
+        //When
+        when(clothingRepository.findById(id))
+                .thenReturn(Mono.just(entity));
+        when(clothingRepository.save(any(ClothingEntity.class)))
+                .thenReturn(Mono.just(saved));
+        when(clothing.getId())
+                .thenReturn(id);
+
+        //Then
+        clothingService.update(clothing)
+                .as(StepVerifier::create)
+                .assertNext(new Consumer<Clothing>() {
```

Comment:
You can use a lambda here to make it cleaner

Reply:
Fixed